### PR TITLE
Weave eval jp

### PR DESCRIPTION
**Pull request by olachinkei**
- Add Japanese evaluation scheme
- Add performance improvement code
- Make evaluation main.py richer to accept more setting
- Update README

### CODE REVIEW

**Review by morganmcg1 — CHANGES_REQUESTED** [review]
Thanks Kei, can you run both JP and EN evaluations on this branch and post the results and links to the weave evals so that we can ensure there is no unexpected change for either language.

**Comment by olachinkei** [comment]
Hi @morganmcg1 , sorry for being late to reply.


"jp v1.2.0" and "en v1.2.0" in the following project are the evaluation results respectively.
https://wandb.ai/wandbot/wandbot-eval-jp/weave/evaluations?sort=%5B%7B%22field%22%3A%22attributes.evaluation_strategy_name%22%2C%22sort%22%3A%22asc%22%7D%5D

The result of evaluation fluctuates, so that of "en v1.2.0" seems to decrease a little bit.

FYI, I newly created jp evaluation dataset by translating the Japanese dataset and eliminating two samples related with the legacy Traces function.

File: README.md [hunk 1]
```diff
@@ -19,12 +19,19 @@ This release introduces a number of exciting updates and improvements:
...
 
+
+Japanese
 | wandbot version  | Comment  | response accuracy |
 |---|---|---|
-| 1.0.0 | our baseline wandbot |  53.78 % |
-| 1.1.0 | improvement over baseline; in production for the longest | 72.45 %  | 
-| 1.2.0 | our new enhanced wandbot | 81.63 % |
+| 1.2.0 | our new enhanced wandbot | 56.3 % |
+| 1.2.1 | add translation process | 71.9 % |
```
Comment (ayulockin) [inline README.md:34]: @parambharat @morganmcg1 do we agree with this nomenclature?

File: README.md [hunk 2]
```diff
@@ -118,8 +126,18 @@ Launch the wandbot with 8 workers. This speeds up evaluation
...
 
-Launch W&B Weave evaluation
 
+
+Set up for evaluation
+
+wandbot/src/wandbot/evaluation/config.py
+- evaluation_strategy_name : attribute name in Weave Evaluation dashboard
+- eval_dataset : 
+    - Latest English evaluation dataset: "weave:///wandbot/wandbot-eval/object/wandbot_eval_data:eCQQ0GjM077wi4ykTWYhLPRpuGIaXbMwUGEB7IyHlFU"
```
Comment (ayulockin) [inline README.md:136]: It would be better to provide the link to the weave dataset webpage along with the actual weave url.

File: README.md [hunk 3]
```diff
@@ -118,8 +126,18 @@ Launch the wandbot with 8 workers. This speeds up evaluation
...
+    - Latest Japanese evaluation dataset: "weave:///wandbot/wandbot-eval-jp/object/wandbot_eval_data_jp:oCWifIAtEVCkSjushP0bOEc5GnhsMUYXURwQznBeKLA"
```
Comment (ayulockin) [inline README.md:137]: same

File: src/wandbot/evaluation/config.py [hunk 4]
```diff
@@ -3,34 +3,21 @@
...
-        env="EVAL_ARTIFACT_ROOT",
-        validation_alias="eval_artifact_root",
+    evaluation_strategy_name: str = Field("jp v1.2.0-beta", description="Will be shown in evaluation page, and be used for just visibility")
+    eval_dataset: str = Field(
+        "weave:///wandbot/wandbot-eval-jp/object/wandbot_eval_data_jp:oCWifIAtEVCkSjushP0bOEc5GnhsMUYXURwQznBeKLA"
+        ,description="Dataset reference for evaluation"
     )
+    # en evaluation dataset: "weave:///wandbot/wandbot-eval/object/wandbot_eval_data:eCQQ0GjM077wi4ykTWYhLPRpuGIaXbMwUGEB7IyHlFU"
+    # jp evaluation dataset: "weave:///wandbot/wandbot-eval-jp/object/wandbot_eval_data_jp:oCWifIAtEVCkSjushP0bOEc5GnhsMUYXURwQznBeKLA" 
+    # jp small evaluation dataset: "weave:///wandbot/wandbot-eval-jp/object/wandbot_eval_data_jp_test:Qp5qAlNYhzLJSfDZONaBNDzjhEmlfOTXNI1NvYhELKQ"
```
Comment (ayulockin) [inline src/wandbot/evaluation/config.py:13]: do we want to keep be commented out?

File: src/wandbot/evaluation/config.py [hunk 5]
```diff
@@ -3,34 +3,21 @@
...
-        validation_alias="eval_output_file",
-    )
     eval_judge_model: str = Field(
         "gpt-4-1106-preview",
         env="EVAL_JUDGE_MODEL",
         validation_alias="eval_judge_model",
     )
     wandb_entity: str = Field("wandbot", env="WANDB_ENTITY")
-    wandb_project: str = Field("wandbot-eval", env="WANDB_PROJECT")
+    wandb_project: str = Field("wandbot-eval-jp", env="WANDB_PROJECT")
```
Comment (ayulockin) [inline src/wandbot/evaluation/config.py:23]: can you leave a note in the README to switch the project name according to the eval strategy?

File: src/wandbot/evaluation/jp_evaluation_dataprep.py [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+        "question": row["question"],
+        "ground_truth": row["answer"],
+        "notes": row["notes"],
+        "context": row["context"],
+        "correctness": row["correctness"],
+        "is_wandb_query": row["is_wandb_query"]
+    } for row in question_rows
+]
+
+def translate_with_claude(text: str) -> str:
```
Comment (ayulockin) [inline src/wandbot/evaluation/jp_evaluation_dataprep.py:21]: `translate_with_openai`

File: src/wandbot/evaluation/weave_eval/main.py [hunk 7]
```diff
@@ -56,6 +66,7 @@ async def get_eval_record(
         "system_prompt": response["system_prompt"],
         "generated_answer": response["answer"],
         "retrieved_contexts": response["source_documents"],
+        "retrieved_contexts_individual": parse_text_to_json(response["source_documents"]),
```
Comment (ayulockin) [inline src/wandbot/evaluation/weave_eval/main.py:69]: remove line 68. no need to add both concatenated and individual contexts.

Comment (olachinkei) [reply src/wandbot/evaluation/weave_eval/main.py:68]: Deleted in the new commit!

File: src/wandbot/evaluation/jp_evaluation_dataprep.py (hunk 6 above)
Comment (olachinkei) [reply src/wandbot/evaluation/jp_evaluation_dataprep.py]: Updated in the new commit!

File: src/wandbot/evaluation/config.py (hunk 5 above)
Comment (olachinkei) [reply src/wandbot/evaluation/config.py]: Updated in the new commit!

File: src/wandbot/evaluation/config.py (hunk 4 above)
Comment (olachinkei) [reply src/wandbot/evaluation/config.py]: Deleted them in the new commit!

**Review by ayulockin — APPROVED** [review]

**Review by morganmcg1 — APPROVED** [review]

**Pull request merged**